packages/dev/core/src/Maths/math.vector.ts

```diff
@@ -1608,13 +1921,40 @@ export class Vector3 {
         return false;
     }
 
+    /**
+     * Gets the current Vector3's floored values and stores them in result
+     * @param result the vector to store the result in
+     * @returns the result vector
+     */
+    public floorToRef<T extends this>(result: T): T {
+        result._x = Math.floor(this._x);
+        result._y = Math.floor(this._y);
+        result._z = Math.floor(this._z);
+        result._isDirty = true;
+        return result;
+    }
+
     /**
      * Gets a new Vector3 from current Vector3 floored values
      * Example Playground https://playground.babylonjs.com/#R1F8YU#22
      * @returns a new Vector3
      */
     public floor(): this {
-        return new (this.constructor as Vector3Constructor<this>)(Math.floor(this._x), Math.floor(this._y), Math.floor(this._z));
+        const result = new (this.constructor as Constructor<typeof Vector3, this>)();

```

Comment:
Why not using the same implementation than in Vector2 (`return new (this.constructor as Constructor<typeof Vector3, this>)(Math.floor(this.x), Math.floor(this.y), Math.floor(this.z));`)?

Reply:
Resolved in 4e38dc4